velox/common/memory/SharedArbitrator.cpp

```diff
@@ -594,46 +673,33 @@ void SharedArbitrator::abort(
   VELOX_CHECK(pool->aborted());
 }
 
-uint64_t SharedArbitrator::decrementFreeCapacity(uint64_t bytes) {
-  uint64_t reserveBytes;
-  uint64_t freeCapacity;
-  {
-    std::lock_guard<std::mutex> l(mutex_);
-    reserveBytes = decrementFreeCapacityLocked(bytes);
-    freeCapacity = freeCapacity_;
-  }
-  RECORD_METRIC_VALUE(kMetricArbitratorFreeCapacityBytes, freeCapacity);
-  return reserveBytes;
-}
-
-uint64_t SharedArbitrator::decrementFreeCapacityLocked(uint64_t bytes) {
-  const uint64_t targetBytes = std::min(freeCapacity_, bytes);
-  VELOX_CHECK_LE(targetBytes, freeCapacity_);
-  freeCapacity_ -= targetBytes;
-  return targetBytes;
-}
-
 void SharedArbitrator::incrementFreeCapacity(uint64_t bytes) {
-  uint64_t freeCapacity;
-  {
-    std::lock_guard<std::mutex> l(mutex_);
-    incrementFreeCapacityLocked(bytes);
-    freeCapacity = freeCapacity_;
-  }
-  RECORD_METRIC_VALUE(kMetricArbitratorFreeCapacityBytes, freeCapacity);
+  std::lock_guard<std::mutex> l(mutex_);
+  incrementFreeCapacityLocked(bytes);
 }
 
 void SharedArbitrator::incrementFreeCapacityLocked(uint64_t bytes) {
-  freeCapacity_ += bytes;
-  if (FOLLY_UNLIKELY(freeCapacity_ > capacity_)) {
+  incrementFreeReservedCapacityLocked(bytes);
+  freeNonReservedCapacity_ += bytes;
+  if (FOLLY_UNLIKELY(
+          freeNonReservedCapacity_ + freeReservedCapacity_ > capacity_)) {
     VELOX_FAIL(
-        "The free capacity {} is larger than the max capacity {}, {}",
-        succinctBytes(freeCapacity_),
+        "The free capacity {}/{} is larger than the max capacity {}, {}",
+        succinctBytes(freeNonReservedCapacity_),
+        succinctBytes(freeReservedCapacity_),
         succinctBytes(capacity_),
         toStringLocked());
   }
 }
 
+void SharedArbitrator::incrementFreeReservedCapacityLocked(uint64_t& bytes) {
```

Comment:
shall we, instead of passing by reference, return the actual incremented bytes to be consistent with other methods in this class?